frontend/packages/ceph-storage-plugin/src/components/create-storage-system/payloads.ts

```diff
@@ -139,3 +141,29 @@ export const createExternalSubSystem = async (subSystemPayloads: Payload[]) => {
     throw err;
   }
 };
+
+const validateCRDReadiness = (crd: K8sResourceKind, plural: string) =>
+  crd?.status?.acceptedNames?.plural === plural;
+
+export const waitforCRD = async (name: string, plural: string, maxAttempts = 15) => {
+  let attempts = 0;
+
+  const pollCRD = async (resolve, reject) => {
+    try {
+      attempts++;
+      const crd = await k8sGet(CustomResourceDefinitionModel, name);
+      const isCrdReady = validateCRDReadiness(crd, plural);
```

Comment:
I am confused how this validates if a CRD is ready? Can you point me to some documentation or K8s resource pages? Thanks.

Reply:
This is what i observed in practice and testing.
And I think its for any k8s resource. The status field should be present which is added once the CRD is created.
If status field is absent then cr creation will throw error

Reply:
I am not sure this is required. Once a CR of kind CRD is created with a particular name shouldn't the presence of that be enough? When we test for CephCluster presence we don't check for it's status field we only check for the presence of a CR of CephCluster kind. The same logic should be valid for CRDs as well.

Reply:
No presence will not be enough, since just after that we are creating cr and it was failing in.
It could result in a not found error

Reply:
This is not new https://github.com/kube-reporting/metering-operator/blob/0d7ecfb5f9ca4603990bb264eae4d998da0e1060/pkg/deploy/install.go#L128-L129